internal/providers/terraform/azure/vpn_gateway.go

```diff
@@ -0,0 +1,171 @@
+	costComponents := make([]*schema.CostComponent, 0)
+
+	if sku == "Basic" {
+		sku = "Basic Gateway"
+	}
+
+	costComponents = append(costComponents, vpnGateway(region, sku))
+
+	if u != nil && u.Get("s2s_tunnel").Type != gjson.Null {
+		tunnel = decimalPtr(decimal.NewFromInt(u.Get("s2s_tunnel").Int()))
+		if tunnel != nil {
+			tunnelLimits := []int{10, 30}
+			tunnelValues := usage.CalculateTierBuckets(*tunnel, tunnelLimits)
+			if tunnelValues[1].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayS2S(region, sku, &tunnelValues[1]))
+			}
+		}
+	} else {
+		costComponents = append(costComponents, vpnGatewayS2S(region, sku, tunnel))
+	}
+
+	if u != nil && u.Get("p2p_connection").Type != gjson.Null {
+		connection = decimalPtr(decimal.NewFromInt(u.Get("p2p_connection").Int()))
+		if connection != nil {
+			connectionLimits := []int{128, 250, 500, 1000, 5000, 10000}
+			connectionValues := usage.CalculateTierBuckets(*connection, connectionLimits)
+			if sku == "VpnGw1" && connectionValues[1].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayP2P(region, sku, &connectionValues[1]))
+			} else if sku == "VpnGw2" && connectionValues[2].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayP2P(region, sku, &connectionValues[2]))
+			} else if sku == "VpnGw3" && connectionValues[3].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayP2P(region, sku, &connectionValues[3]))
+			} else if sku == "VpnGw4" && connectionValues[4].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayP2P(region, sku, &connectionValues[4]))
+			} else if sku == "VpnGw5" && connectionValues[5].GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, vpnGatewayP2P(region, sku, &connectionValues[5]))
+			}
+		}
+	} else {
+		costComponents = append(costComponents, vpnGatewayP2P(region, sku, connection))
+	}
+
+	if u != nil && u.Get("data_transfers").Type != gjson.Null {
+		dataTransfers = decimalPtr(decimal.NewFromInt(u.Get("data_transfers").Int()))
+		if dataTransfers != nil {
+			costComponents = append(costComponents, vpnGatewayDataTransfers(zone, sku, dataTransfers))
+		}
+	} else {
+		costComponents = append(costComponents, vpnGatewayDataTransfers(zone, sku, dataTransfers))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+	}
+}
+
+func vpnGateway(region, sku string) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:           fmt.Sprintf("VPN Gateway (%s)", sku),
+		Unit:           "hours",
+		UnitMultiplier: decimal.NewFromInt(1),
+		HourlyQuantity: decimalPtr(decimal.NewFromInt(1)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName: strPtr("azure"),
+			Region:     strPtr(region),
+			Service:    strPtr("VPN Gateway"),
+			AttributeFilters: []*schema.AttributeFilter{
+				{Key: "meterName", ValueRegex: strPtr(fmt.Sprintf("/%s/i", sku))},
```

Comment:
Do we need to add start/end (`/^%s$/i`anchors to these sku regexes? Some of the skus 'overlap' so looking up `/VpnGw3/` will match VpnGw3 and VpnGw3AZ and I believe we would get multiple prices.